apis/v1beta1/vspherevm_types.go

```diff
@@ -40,6 +40,143 @@ const (
+)
+
+// VSphereVM's IPAddressClaimsFulfilled condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// VSphereVMIPAddressClaimsFulfilledV1Beta2Condition documents the status of claiming an IP address
+	// from an IPAM provider.
+	VSphereVMIPAddressClaimsFulfilledV1Beta2Condition = "IPAddressClaimsFulfilled"
+
+	// VSphereVMIPAddressClaimsBeingCreatedV1Beta2Reason documents that claims for the
+	// IP addresses required by the VSphereVM are being created.
+	VSphereVMIPAddressClaimsBeingCreatedV1Beta2Reason = "IPAddressClaimsBeingCreated"
+
+	// VSphereVMIPAddressClaimsWaitingForIPAddressV1Beta2Reason documents that claims for the
+	// IP addresses required by the VSphereVM are waiting for IP addresses.
+	VSphereVMIPAddressClaimsWaitingForIPAddressV1Beta2Reason = "WaitingForIPAddress"
+
+	// VSphereVMIPAddressClaimsFulfilledV1Beta2Reason documents that claims for the
+	// IP addresses required by the VSphereVM are fulfilled.
+	VSphereVMIPAddressClaimsFulfilledV1Beta2Reason = "Fulfilled"
+
+	// VSphereVMIPAddressClaimsNotFulfilledV1Beta2Reason documents that claims for the
+	// IP addresses required by the VSphereVM are not fulfilled.
+	VSphereVMIPAddressClaimsNotFulfilledV1Beta2Reason = "NotFulfilled"
+)
+
+// VSphereVM's GuestSoftPowerOffSucceeded condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// VSphereVMGuestSoftPowerOffSucceededV1Beta2Condition documents the status of performing guest initiated
+	// graceful shutdown.
+	VSphereVMGuestSoftPowerOffSucceededV1Beta2Condition string = "GuestSoftPowerOffSucceeded"
+
+	// VSphereVMGuestSoftPowerOffInProgressV1Beta2Reason documents that the guest receives
+	// a graceful shutdown request.
+	VSphereVMGuestSoftPowerOffInProgressV1Beta2Reason = "InProgress"
+
+	// GuestSoftPowerOffFailedV1Beta2Reason documents that the graceful
+	// shutdown request fails.
+	VSphereVMGuestSoftPowerOffFailedV1Beta2Reason = "Failed"
+
+	// GuestSoftPowerOffSucceededV1Beta2Reason documents that the graceful
+	// shutdown request fails.
+	VSphereVMGuestSoftPowerOffSucceededV1Beta2Reason = "Succeeded"
+)
+
+// VSphereVM's PCIDevicesDetached condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// VSphereVMPCIDevicesDetachedConditionV1Beta2Condition documents the status of the attached PCI devices on the VSphereVM.
+	// It is a negative condition to notify the user that the device(s) is no longer attached to
+	// the underlying VM and would require manual intervention to fix the situation.
+	VSphereVMPCIDevicesDetachedConditionV1Beta2Condition string = "PCIDevicesDetached"
+
+	// VSpherePCIDevicesDetachedNotFoundReasonV1Beta2Reason documents the VSphereVM not having the PCI device attached during VM startup.
+	// This would indicate that the PCI devices were removed out of band by an external entity.
+	VSpherePCIDevicesDetachedNotFoundReasonV1Beta2Reason = "NotFound"
```

Comment:
```suggestion
	// VSpherePCIDevicesDetachedNotFoundV1Beta2Reason documents the VSphereVM not having the PCI device attached during VM startup.
	// This would indicate that the PCI devices were removed out of band by an external entity.
	VSpherePCIDevicesDetachedNotFoundV1Beta2Reason = "NotFound"
```